Handle tile render errors

scytacki opened the pull request:
We were getting CLUE crashes when tiles had rendering errors. Since these errors can happen in the thumbnails, without error handling it is hard to find them. 

A React error boundary is used to catch these errors at the document level. We could add an additional error boundary at the tile level so then the document could still be rendered and just the specific tile would show the error. However the whole document approach seems better to start.

Also a new ErrorTest tile is added to make it easier to demo and test this error handling.

You can see the error page with this with this link:
https://collaborative-learning.concord.org/branch/improve-error-handling/doc-editor.html?unit=./curriculum/example-curriculum/error-unit.json&document=curriculum/example-curriculum/error-document.json

kswenson commented on src/register-tile-types.ts:
```diff
@@ -17,6 +17,7 @@ const gTileRegistration: Record<string, () => void> = {
     import(/* webpackChunkName: "SharedVariables" */"./plugins/shared-variables/shared-variables-registration")
   ]),
   "Drawing": () => import(/* webpackChunkName: "Drawing" */"./plugins/drawing/drawing-registration"),
+  "ErrorTest": () => import(/* webpackChunkName: "Drawing" */"./plugins/error-test/error-test-registration"),
```
```suggestion
  "ErrorTest": () => import(/* webpackChunkName: "Error" */"./plugins/error-test/error-test-registration"),
```

kswenson reviewed (approved):
👍 LGTM

scytacki commented:
I added a QA review on this, so the QA group would be aware of this new feature. 
I added a cypress test to make sure the error page works. 

It could be useful in the future, to setup cypress tests which add an error document to all of the places that documents can be listed in CLUE. But all of these places should be using the same canvas component so they all ought to work.

tejal-shah commented:
Thank you @scytacki , I've created [#186694591](https://www.pivotaltracker.com/story/show/186694591) for the cypress test you mention but I'm not very clear on what you mean by `But all of these places should be using the same canvas component so they all ought to work.` Can you please elaborate in the story I created so whoever who works on the story knows these details? Thanks!

tejal-shah reviewed (approved):
LGTM 👍
